lib/debug/repl.cpp

```diff
@@ -493,6 +515,53 @@ void Repl::addAccelerometerCommands(const std::uint8_t bus, const std::uint8_t d
   addCommand(accelerometer_read_command);
 }
 
+void Repl::addGyroscopeCommands(const std::uint8_t bus, const std::uint8_t device_address)
+{
+  const auto optional_i2c = io::HardwareI2c::create(logger_, bus);
+  if (!optional_i2c) {
+    logger_.log(core::LogLevel::kFatal, "Failed to create I2C instance on bus %d", bus);
+    return;
+  }
+  const auto i2c                = std::move(*optional_i2c);
+  const auto optional_gyroscope = sensors::Gyroscope::create(logger_, i2c, bus, device_address);
+  const auto gyroscope          = std::make_shared<sensors::Gyroscope>(*optional_gyroscope);
+  Command gyroscope_read_command;
+  std::stringstream identifier;
+  identifier << "gyroscope 0x" << std::hex << static_cast<int>(device_address) << " read";
+  gyroscope_read_command.name = identifier.str();
+  std::stringstream description;
+  description << "Read gyroscope sensor 0x" << std::hex << static_cast<int>(device_address)
+              << " on "
+              << "I2C bus " << static_cast<int>(bus);
+  gyroscope_read_command.description = description.str();
+  gyroscope_read_command.handler     = [this, gyroscope, bus]() {
+    core::Float Angle = 0;
+    core::Float Old_data = 0;
+    auto Old_time = std::chrono::system_clock::now();
+    std::optional<core::GyroscopeData> value;
+    
+    while (true) {
+      std::optional<core::GyroscopeData> value = gyroscope->read(core::Axis::kX);
+      const auto result = value.value();
+
+      std::chrono::duration<double> elapsed_seconds = result.measured_at - Old_time;
```

Comment:
Have a look in core. Most of that functionality should already be implemented and if possible, we want to keep the ugly chrono notation contained in there.